File: packages/webservice/models/src/main/kotlin/com/yahoo/navi/ws/models/beans/fragments/layout/Layout.kt
/**
 * Copyright 2020, Yahoo Holdings Inc.
 * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
 */
package com.yahoo.navi.ws.models.beans.fragments.layout

data class Layout(
    var column: Int = 0,
    var row: Int = 0,
    var height: Int = 0,
    var width: Int = 0,
    var widgetId: Int = 0
)

File: packages/core/addon/models/fragments/layout.ts
/**
 * Copyright 2021, Yahoo Holdings Inc.
 * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
 */
import attr from 'ember-data/attr';
import Fragment from 'ember-data-model-fragments/fragment';

export default class LayoutFragment extends Fragment {
  @attr('number') column!: number;
  @attr('number') row!: number;
  @attr('number') width!: number;
  @attr('number') height!: number;
  @attr('number') widgetId!: number;
}

declare module '../registry' {
  export interface FragmentRegistry {
    'fragments/layout': LayoutFragment;
  }
}
